contracts/v3/strategies/MIMConvexStrategy.sol

```diff
@@ -0,0 +1,153 @@
+     */
+    constructor(
+        string memory _name,
+        address _want,
+        address _crv,
+        address _cvx,
+        address _weth,
+        address _mim,
+        address _crv3,
+        uint256 _pid,
+        IConvexVault _convexVault,
+        IStableSwap2Pool _stableSwap2Pool,
+        address _controller,
+        address _manager,
+        address _router
+    ) public BaseStrategy(_name, _controller, _manager, _want, _weth, _router) {
+        require(address(_crv) != address(0), '!_crv');
+        require(address(_cvx) != address(0), '!_cvx');
+        require(address(_mim) != address(0), '!_mim');
+        require(address(_crv3) != address(0), '!_crv3');
+        require(address(_convexVault) != address(0), '!_convexVault');
+        require(address(_stableSwap2Pool) != address(0), '!_stableSwap2Pool');
+
+        (, address _token, , address _crvRewards, , ) = _convexVault.poolInfo(_pid);
+        crv = _crv;
+        cvx = _cvx;
+        mim = _mim;
+        crv3 = _crv3;
+        pid = _pid;
+        convexVault = _convexVault;
+        mimCvxDepositLP = _token;
+        crvRewards = IConvexRewards(_crvRewards);
+        stableSwap2Pool = _stableSwap2Pool;
+        // Required to overcome "Stack Too Deep" error
+        _setApprovals(
+            _want,
+            _crv,
+            _cvx,
+            _mim,
+            _crv3,
+            address(_convexVault),
+            address(_stableSwap2Pool)
+        );
+    }
+
+    function _setApprovals(
+        address _want,
+        address _crv,
+        address _cvx,
+        address _mim,
+        address _crv3,
+        address _convexVault,
+        address _stableSwap2Pool
+    ) internal {
+        IERC20(_want).safeApprove(address(_convexVault), type(uint256).max);
+        IERC20(_crv).safeApprove(address(router), type(uint256).max);
+        IERC20(_cvx).safeApprove(address(router), type(uint256).max);
+        IERC20(_mim).safeApprove(address(_stableSwap2Pool), type(uint256).max);
+        IERC20(_crv3).safeApprove(address(_stableSwap2Pool), type(uint256).max);
+        IERC20(_want).safeApprove(address(_stableSwap2Pool), type(uint256).max);
+    }
+
+    function _deposit() internal override {
+        if (balanceOfWant() > 0) {
+            convexVault.depositAll(pid, true);
+        }
+    }
+
+    function _claimReward() internal {
+        crvRewards.getReward(address(this), true);
+    }
+
+    function _addLiquidity() internal {
+        uint256[2] memory amounts;
+        amounts[0] = IERC20(mim).balanceOf(address(this));
+        amounts[1] = IERC20(crv3).balanceOf(address(this));
+        stableSwap2Pool.add_liquidity(amounts, 1);
+    }
+
+    function getMostPremium() public view returns (address, uint256) {
+        // both MIM and 3CRV have 18 decimals
+        if (stableSwap2Pool.balances(0) > stableSwap2Pool.balances(1)) {
+            return (crv3, 1);
+        }
+
+        return (mim, 0); // If they're somehow equal, we just want MIM
```

Comment:
This comment is not correct, is it?

Reply:
yeah. It is not very accurate. I can change it to 

// Otherwise, we just want MIM